File: chapter_2/only_repeat_num.py
def repeated_number(nums):
    for i in range(0, len(nums)):
        while nums[i] != i+1:
            if nums[nums[i]-1] == nums[i]:
                return nums[i]
            else:
                nums[nums[i]-1], nums[i] = nums[i], nums[nums[i]-1]

if __name__ == "__main__":
    print(repeated_number([4, 1, 5, 9, 2, 8, 6, 7, 2, 3]))

File: chapter_2/only_repeat_num.rs
fn repeated_number(nums: &mut [usize]) -> &usize {
    for i in 0..nums.len() {
        while nums[i] != i+1 {
            if nums[nums[i]-1] == nums[i] {
                return &nums[i];
            } else {
                nums.swap(nums[i]-1, i);
            }   
        }
    }
    &0
}

fn main() {
    println!("{}", repeated_number(&mut [4, 1, 5, 9, 3, 8, 6, 7, 2, 3]));
}